src/test/java/org/isf/medicalstock/test/Tests.java

```diff
@@ -73,108 +87,214 @@
-	@Autowired
-	MedicalsIoOperationRepository medicalsIoOperationRepository;
-	@Autowired
-	MedicalTypeIoOperationRepository medicalTypeIoOperationRepository;
-	@Autowired
-	WardIoOperationRepository wardIoOperationRepository;
-	@Autowired
-	MovementIoOperationRepository movementIoOperationRepository;
-	@Autowired
-	MedicalStockMovementTypeIoOperationRepository medicalStockMovementTypeIoOperationRepository;
-	@Autowired
-	SupplierIoOperationRepository supplierIoOperationRepository;
-	@Autowired
-	ApplicationEventPublisher applicationEventPublisher;
+	private EntityManager entityManager;
 
-	@BeforeClass
-	public static void setUpClass() {
-		jpa = new DbJpaUtil();
-		testLot = new TestLot();
-		testMovement = new TestMovement();
-		testMedical = new TestMedical();
-		testMedicalType = new TestMedicalType();
-		testMovementType = new TestMovementType();
-		testWard = new TestWard();
-		testSupplier = new TestSupplier();
-	}
+    @Autowired
+    MedicalStockIoOperations medicalStockIoOperation;
 
-	@Before
-	public void setUp() throws OHException {
-		cleanH2InMemoryDb();
-		jpa.open();
-		testLot.setup(false);
-	}
+    public Tests(boolean in, boolean out, boolean toward) {
+	    GeneralData.AUTOMATICLOT_IN = in;
+	    GeneralData.AUTOMATICLOT_OUT = out;
+	    GeneralData.AUTOMATICLOTWARD_TOWARD = toward;
+    }
 
-	@After
-	public void tearDown() throws Exception
-	{
-		jpa.flush();
-		jpa.close();
+	@BeforeClass
+    public static void setUpClass()  
+    {
+    	jpa = new DbJpaUtil();
+    	testLot = new TestLot();
+    	testMovement = new TestMovement();
+    	testMedical = new TestMedical();
+    	testMedicalType = new TestMedicalType();
+    	testMovementType = new TestMovementType();
+    	testWard = new TestWard();
+    	testSupplier = new TestSupplier();
+    }
+
+    @Before
+    public void setUp() throws OHException
+    {
+        jpa.open();
+        
+        testLot.setup(false);
+    }
+        
+    @After
+    public void tearDown() throws Exception 
+    {
+    	cleanH2InMemoryDb();
+        jpa.close();
+    }
+    
+    @AfterClass
+    public static void tearDownClass() throws OHException 
+    {
+    	testLot = null;
+    	testMovement = null;
+    	testMedical = null;
+    	testMedicalType = null;
+    	testMovementType = null;
+    	testWard = null;
+    	testSupplier = null;
+    }
+
+	@Parameterized.Parameters(name ="Test with AUTOMATICLOT_IN={0}, AUTOMATICLOT_OUT={1}, AUTOMATICLOTWARD_TOWARD={2}")
+	public static Collection automaticlot() {
+		return Arrays.asList(new Object[][] {
+				{ false, false, false },
+				{ false, false, true },
+				{ false, true, false },
+				{ false, true, true },
+				{ true, false, false },
+				{ true, false, true },
+				{ true, true, false },
+				{ true, true, true }
+		});
 	}
 
 	@Test
-	public void testLotGets() throws Exception {
-		String code = _setupTestLot(false);
-		_checkLotIntoDb(code);
+	public void testLotGets() 
```

Comment:
You must be using the wrong branch as all this try-catch-fail code is gone.

Reply:
Added Exception